### PR TITLE
GUVNOR-2544: Guided Decision Table Editor: Layout manager for tables.

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-2544

### CODE REVIEW

**Comment by manstis** [comment]
Please also see https://github.com/droolsjbpm/drools-wb/pull/206

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/GridData.java [hunk 1]
```diff
@@ -135,6 +135,19 @@ void moveRowsTo( final int index,
     int getRowCount();
 
     /**
+     * Returns the number of rows in the grid's header.
+     * @return
+     */
```
Comment (csadilek) [inline uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/GridData.java]: empty ```@return```

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/GridData.java [hunk 2]
```diff
@@ -135,6 +135,19 @@ void moveRowsTo( final int index,
...
+    int getHeaderRowCount();
+
+    /**
+     * Sets the number of rows in the grid's header.
+     * @param headerRowCount Cannot be less than 1.
+     * @return
+     */
```
Comment (csadilek) [inline uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/GridData.java]: ```@return``` can be removed as method returns void

**Comment by csadilek** [comment]
+1 (some minor javadoc nitpicking)

**Pull request merged**